contracts/access/manager/IAuthority.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+interface IAuthority {
+    function canCall(address caller, address target, bytes4 selector) external view returns (bool allowed);
```

Comment:
> a `canCall` call will fail for anyone using this interface and calling the version returning a tuple.

It's the other way around. If we define it here to return `(bool allowed)`, calling `AccessManager.canCall` will succeed, the extra return value is ignored.

On the other hand, if we define it here to return `(bool allowed, uint32 delay)`, calling this function on an Authority that only returns `allowed` will revert, because the return data is not large enough to decode two values.

Reply:
Ok, then I think we need to document this trick somehow, maybe for another PR (perhaps opening an issue).

I think it's not immediately evident the mechanics behind the `canCall` decoding. What do you think?

Reply:
I think this might be sufficiently documented now in the AuthorityUtils library.